internal/pkg/agent/application/upgrade/step_unpack.go

```diff
@@ -242,3 +358,110 @@ func validFileName(p string) bool {
 	}
 	return true
 }
+
+type pathMapper struct {
+	mappings []map[string]string
+}
+
+func (pm pathMapper) Map(path string) string {
+	for _, mapping := range pm.mappings {
+		for pkgPath, mappedPath := range mapping {
+			if strings.HasPrefix(path, pkgPath) {
+				return filepath.Join(mappedPath, path[len(pkgPath):])
+			}
+		}
+	}
+	return path
+}
+
+type tarCloser struct {
+	tarFile    *os.File
+	gzipReader *gzip.Reader
+}
+
+func (tc *tarCloser) Close() error {
+	var err error
+	if tc.gzipReader != nil {
+		err = multierror.Append(err, tc.gzipReader.Close())
```

Comment:
nit: couldn't this be errors.Join()?

Reply:
it can be, this was taken from the existing code which used `multierror` package. I will change it

Reply:
done in 9945a142d817b9ae2c21437e6b5a9cb813d61801